spec/lib/approval_request_listener_spec.rb

```diff
@@ -0,0 +1,124 @@
+describe ApprovalRequestListener do
+  let(:client) { double(:client) }
+  let(:event)  { ApprovalRequestListener::EVENT_REQUEST_FINISHED }
+
+  describe "#subscribe_to_approval_updates" do
+    let(:request) do
+      { :headers => { 'x-rh-identity' => encoded_user_hash }, :original_url => 'whatever' }
+    end
+
+    around do |example|
+      ManageIQ::API::Common::Request.with_request(request) { example.call }
+    end
+
+    let(:messages) { [ManageIQ::Messaging::ReceivedMessage.new(nil, event, payload, nil)] }
+    let!(:approval_request) do
+      ApprovalRequest.create!(
+        :approval_request_ref => approval_request_ref,
+        :workflow_ref         => "2323",
+        :order_item_id        => "3434"
+      )
+    end
+
+    before do
+      allow(ManageIQ::Messaging::Client).to receive(:open).with(
+        :protocol   => :Kafka,
+        :client_ref => ApprovalRequestListener::CLIENT_AND_GROUP_REF,
+        :group_ref  => ApprovalRequestListener::CLIENT_AND_GROUP_REF
+      ).and_return(client)
+      allow(client).to receive(:subscribe_topic).with(
+        :service   => ApprovalRequestListener::SERVICE_NAME,
+        :max_bytes => 500_000
+      ).and_yield(messages)
+      allow(client).to receive(:close)
+    end
+
+    context "when the approval request is not findable" do
+      let(:approval_request_ref) { "0" }
+
+      context "when the request_id is anything else" do
+        let(:payload)    { {"request_id" => request_id } }
+        let(:request_id) { "10" }
```

Comment:
Is there any reason to separate this out? It's not being used elsewhere in the context, we can just put `"10"` in the previous line.

Alternatively, since this is the only one where the payload doesn't have the "decision" and "reason" keys, you could move the `payload` `let` up a few contexts and leave this one here and remove the others from the other contexts. Then it would make more sense to have the `request_id` `let`s here and down there. You could also move this one out if having the decision and reason keys doesn't affect this spec and put in dummy values for this context.

Your call, but I think any of those options would clean it up a bit.